chat-commands.js

```diff
@@ -2374,24 +2374,27 @@ exports.commands = {
 	hidealttext: 'hidetext',
 	hidealtstext: 'hidetext',
 	hidetext: function (target, room, user, connection, cmd) {
-		if (!target) return this.parse(`/help hidetext`);
+		if (!target) return this.parse('/help hidetext');
+		if (!room.log.hasUsername(target)) return this.errorReply(`User ${target} not found or has no roomlogs.`);
```

Comment:
This is, incidentally, the wrong place to check this.

Unless you want to make your code especially fancy, you should probably only check this if `targetUser` is not found.

Reply:
checking this after the ``targetUser`` check doesn't work. I think this is the best place as the command should stop immediately if the target user has no lines in the chat.

Reply:
It's possible that the target user has no lines in the chat because they were using a different username.

Reply:
Hmm I’ll have to hold off fixing this last thing for a few hours as I have some running around I have to do. Although thinking about it, wouldn’t it be better for ``hasUsername();`` check for alts?

Reply:
>checking this after the targetUser check doesn't work.

>wouldn’t it be better for hasUsername() check for alts?

... how would it do that for users with expired user objects?

Reply:
Have you read the ``hasUsername()`` code?

Reply:
Oh I guess my comment was sort of vague, I tried originally doing this after the ``!targetUser && user.can(‘lock’)`` check but doing something like ``!targetUser && !room.log.hasUsername(target)`` wasn’t working either